CONTRIBUTING.md

```diff
@@ -63,6 +63,13 @@ each other and from whatever else you are doing with your repository!
 
 You must [GPG-sign](./contributing/signing.rst) your work, certifying that you either wrote the work or otherwise have the right to pass it on to an open-source project. This is *not* just a "signed-off-by" commit, but instead, a digitally signed one.
 
+### Breaking Changes
+
+In general any changes that would disrupt existing uses of the framework are considered "breaking changes" and are not allowed. A few specific examples to pay attention to:
+
+1. New classes/properties/constants in `system` are acceptable, but anything in the `app` directory that will be used in `system` should be backwards-compatible.
+2. Any changes to non-private methods must be backwards-compatible with the original definition.
+
```

Comment:
More additions:
3. Deleting non-private properties or methods without prior deprecation notices is frowned upon and will likely be closed.
4. Deleting or renaming public classes and interfaces, as well as those not marked as `@internal`, without prior deprecation notices or not providing fallback solutions will also not be favorably considered.

Reply:
applied.